integration-cli/docker_cli_build_test.go

```diff
@@ -4822,3 +4823,52 @@ func TestBuildVolumeFileExistsinContainer(t *testing.T) {
 
 	logDone("build - errors when volume is specified where a file exists")
 }
+
+func TestBuildMissingArgs(t *testing.T) {
+	// Test to make sure that all Dockerfile commands (except the ones listed
+	// in skipCmds) will generate an error if no args are provided.
+	// Note: INSERT is deprecated so we exclude it because of that.
+
+	skipCmds := map[string]struct{}{
+		"CMD":        {},
+		"RUN":        {},
+		"ENTRYPOINT": {},
+		"INSERT":     {},
+	}
+
+	defer deleteAllContainers()
+
+	for cmd := range builder.EvaluateTable {
+		var dockerfile string
+
+		cmd = strings.ToUpper(cmd)
+
+		if _, ok := skipCmds[cmd]; ok {
+			continue
+		}
+
+		if cmd == "FROM" {
+			dockerfile = cmd
+		} else {
+			// Add FROM to make sure we don't complain about it missing
+			dockerfile = "FROM busybox\n" + cmd
+		}
+
+		ctx, err := fakeContext(dockerfile, map[string]string{})
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer ctx.Close()
+		var out string
+		if out, err = buildImageFromContext("args", ctx, true); err == nil {
+			t.Fatalf("%s was supposed to fail. Out:%s", cmd, out)
+		}
+		if !strings.Contains(err.Error(), cmd+" requires") {
+			t.Fatalf("%s returned the wrong type of error:%s", cmd, err)
+		}
+
+		ctx.Close()
```

Comment:
Not 100% sure of the correct 'go' way to do this, but the `ctx.Close()` here is meant to close the ctx down after each command, as a clean-up.  The defer is there in case we exit prematurely before we get to it. So, this one will negate the need for the defer but is there a way to undo the defer?  Or would it be ok to write it as:
```
out, err = buildImageFromContext("args", ctx, true)
ctx.Close()
if err == nil {
    t.Fatalf("%s was supposed to fail. Out:%s", cmd, out)
}
```
and assume that buildImageFromContext() won't ever panic?  Didn't think that was safe.  So, while (in the happy path cases) the defer is a noop it seemed safer.  Or would you suggest I remove the clean-up one and just do it all in a defer?  Felt odd to have 39 (or so) defers built up.